lib/gh-badges.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+const { makeBadge } = require('./make-badge')
+const { PDFKitTextMeasurer, QuickTextMeasurer } = require('./text-measurer')
+
+class BadgeFactory {
+  constructor({ fontPath, fallbackFontPath, cache = true }) {
+    this.measurer = cache
+      ? new QuickTextMeasurer(fontPath, fallbackFontPath)
+      : new PDFKitTextMeasurer(fontPath, fallbackFontPath)
+  }
+
+  /**
+   * Create a badge
+   *
+   * @param {object} format - Object specifying badge data
+   * @param {string[]} format.text
+   * @param {string} format.colorscheme
+   * @param {string} format.colorA
+   * @param {string} format.colorB
+   * @param {string} format.format
+   * @param {string} format.template
+   * @return {string} Badge in SVG or JSON format
```

Comment:
I'd love to switch this over to the schema we're using for new-style services, though that could definitely happen another day.

Reply:
The DocBlock is helpful, because it allows editors/IDEs to show inline help, but we could definitely improve the validation. Lets save that for a minor release though